tests/nested/core20/kernel-reseal/task.yaml

```diff
@@ -2,19 +2,37 @@ summary: Check that a kernel refresh reseals
 
 systems: [ubuntu-20.04-64]
 
-execute: |
+prepare: |
   # shellcheck source=tests/lib/nested.sh
   . "$TESTSLIB/nested.sh"
 
-  # Wait for snapd to be seeded
-  nested_exec sudo snap wait system seed.loaded
+  snap download pc-kernel --channel=20/edge --basename=pc-kernel
+  unsquashfs -d pc-kernel pc-kernel.snap
+  sed -i 's/This program cannot be run in DOS mode/This program cannot be run in D0S mode/' pc-kernel/kernel.efi
```

Comment:
I'm not sure how (or if ever) we would change the headers in the kernel.efi, but perhaps a `grep -q -a "This program cannot be run in DOS mode" pc-kernel/kernel.efi` before doing the sed would be helpful, otherwise if the sed fails to apply then sed still exits 0, so the test would start failing and we would never know from looking at logs that this replacement just starts silently failing